eng/pipelines/coreclr/templates/build-job.yml

```diff
@@ -181,14 +181,13 @@ jobs:
           df -h
         displayName: Disk Usage before Build
 
-    # Build DacTableGen (Windows-only)
-    - ${{ if eq(parameters.osGroup, 'windows') }}:
-      - script: $(Build.SourcesDirectory)$(dir)build$(scriptExt) -subset clr.dactools $(crossArg) -arch $(archType) $(osArg) -c $(buildConfig) $(officialBuildIdArg) -ci /bl:$(Build.SourcesDirectory)artifacts/log/$(buildConfig)/DacTools.binlog
-        displayName: Build DAC utility tools
+    # Build/Generate native prerequisites
+    - script: $(Build.SourcesDirectory)$(dir)build$(scriptExt) -subset clr.nativeprereqs $(crossArg) -arch $(archType) $(osArg) -c $(buildConfig) $(officialBuildIdArg) -ci /bl:$(Build.SourcesDirectory)artifacts/log/$(buildConfig)/CoreCLRNativePrereqs.binlog
+      displayName: Build and generate native prerequisites
```

Comment:
Why not use the common msbuild entry point to build the CoreCLR runtime and by that avoid the msbuild invocation here and the extra AzDO specific `OutputPgoPathForCI` target?

Reply:
I spoke with @jkotas before doing this work and he said that he still doesn't want us to call into CMake/Ninja from MSBuild if we can avoid it.

Reply:
Sharing my thoughts from my conversation with Jeremy:
- Designs where one build driver (msbuild) is wrapping another build driver (ninja) are very hard to debug.
- I think it is important to have a reasonable way to build the unmanaged runtime without a working managed runtime for bootstrapping on new platforms.
- Things like version files are non-essential and it is easy to replace them with placeholder files. I do not see a problem with using C# to generate them. I would actually prefer to use C# to generate them. It would be nice to get rid of the python dependency.

Reply:
>     * It would be nice to get rid of the python dependency.

For eventpipe generator as well (can be replaced with C++ shim/macros as done for numa and openssl shims). We could also try to get rid of perl dependency, we have some headers generators still written in perl (which not many folks use in this day and age).. 🙂